Mathlib/RingTheory/Idempotents.lean

```diff
@@ -335,6 +345,143 @@ theorem CompleteOrthogonalIdempotents.of_ker_isNilpotent_of_isMulCentral
     simpa [RingHom.mem_ker, sub_eq_zero] using congr_fun h₂.symm i
   exact h₁
 
+omit [Fintype I] in
+lemma ne_sum_ortho {e : R} (he : IsPrimitiveIdempotents e)
+    (e' : I → R) (ho : OrthogonalIdempotents (R := R) e')
+    (_ : ∀ i, e' i ≠ 0) (i j : I) :
+    e ≠ e' i + e' j := by
+  have : e ≠ 0 := he.ne_zero
+  by_cases eq : i = j
+  · subst eq
+    intro rid
+    have eq : e * e = e := he.idem
+    conv_lhs at eq => rw [rid]
+    simp only [mul_add, add_mul] at eq
+    rw [ho.idem i, ← rid] at eq
+    simp only [add_right_eq_self] at eq
+    contradiction
+  · apply he.ne_sum_ortho' (e' i) (e' j)
+      ⟨fun x ↦ by
+        simp only [Nat.succ_eq_add_one, Nat.reduceAdd]
+        fin_cases x <;> simp [Nat.succ_eq_add_one, Nat.reduceAdd, Fin.zero_eta, Fin.isValue,
+            Matrix.cons_val_zero, Matrix.cons_val_one, ho.idem],
+      fun x y hxy ↦ by
+        fin_cases x <;> fin_cases y <;> simp_all [ho.2 eq]
+        exact ho.2 <| fun a ↦ eq (a.symm)⟩
+    · simp_all only [ne_eq, not_false_eq_true]
+    · simp_all only [ne_eq, not_false_eq_true]
+
+variable {ι M σ : Type*} [DecidableEq ι] [AddCommGroup M] [Module R M]
+open DirectSum
+variable (ℳ : ι → Submodule R M) [Decomposition ℳ] [(i : ι) → (x : (ℳ i)) → Decidable (x ≠ 0)]
+
+lemma decompose_unique  (rep₁ rep₂ : ⨁ i, ℳ i)
```

Comment:
This lemma belongs to `Mathlib.Algebra.DirectSum.Decomposition` or something like that, and probably can be generalized

Reply:
Done!